packages/twenty-front/src/modules/app/effect-components/PageChangeEffect.tsx

```diff
@@ -38,6 +45,11 @@ export const PageChangeEffect = () => {
 
   const eventTracker = useEventTracker();
 
+  const objectNamePlural =
```

Comment:
@lucasbordeau `useResetTableRowSelection(objectNamePlural)` seems to throw whenever `objectNamePlural` is an empty string because of [this](https://github.com/twentyhq/twenty/blob/4f7fa7adcdf9598c94a4806e678b48a1cc5e3da0/packages/twenty-front/src/modules/object-record/record-table/hooks/internal/useResetTableRowSelection.ts#L16C1-L19C5).

But if we're setting the reset logic in this `PageChangeEffect`, since hooks can't be called conditionally, every time the code reaches this point:

```
const resetTableSelections = useResetTableRowSelection(objectNamePlural);
```

and the user is not currently in the `RecordIndexPage`, this is bound to throw. I've currently used `CoreObjectNamePlural.Person` as a workaround, but don't think this the right way to do it. How would you handle it?

Reply:
This is technical debt at this point, we have to refactor this hook. 

I suggest that you add a TODO comment and let your current fix which is enough for now.